### PR TITLE
BACKLOG-12608 : use selection to handle one item selection with the e…

### PR DESCRIPTION
**Pull request by Yousria**
…dit action

## JIRA

https://jira.jahia.org/browse/BACKLOG-12608

## Tests

The following are included in this PR
- [ ] Unit Tests (Most changes _should_ have unit tests)
- [ ] Integration Tests

## Checklist

I have considered the following implications of my change: 

- [ ] Security (in particular for changes to authentication, authorization, data fetching, ...)
- [ ] Performance
- [ ] Migration
- [ ] Code maintainability

## Documentation

- [ ] Inline documentation
- [ ] Internal Documentation (wiki)
- [ ] User-facing Documentation

### CODE REVIEW

File: src/javascript/JContent/ContentRoute/MainActionBar/MainActionBar.jsx [hunk 1]
```diff
@@ -5,18 +5,18 @@ import {DisplayAction} from '@jahia/ui-extender';
...
 
-const ButtonRenderer = getButtonRenderer();
 const ButtonRendererShortLabel = getButtonRenderer({labelStyle: 'short'});
 const ButtonRendererNoLabel = getButtonRenderer({labelStyle: 'none'});
 
 export const MainActionBar = () => {
-    const {path} = useSelector(state => ({path: state.jcontent.path}));
-
+    const {path, selection} = useSelector(state => ({path: state.jcontent.path, selection: state.jcontent.selection}));
+    let editPath = (selection && selection.length === 0) ? path : selection[0];
```
Comment (tdraier) [inline src/javascript/JContent/ContentRoute/MainActionBar/MainActionBar.jsx]: Why taking the selection ? I can't see anything in description or mockup about that ?

File: src/javascript/JContent/ContentRoute/MainActionBar/MainActionBar.jsx [hunk 2]
```diff
@@ -5,18 +5,18 @@ import {DisplayAction} from '@jahia/ui-extender';
...
+    let enabled = selection.length < 2;
     return (
         <div className={styles.root}>
             <Separator variant="vertical"/>
-            <DisplayAction actionKey="pageComposer" context={{path}} render={ButtonRenderer} size="big" variant="ghost" color="accent" className={styles.item}/>
-            <DisplayAction actionKey="edit" context={{path}} render={ButtonRenderer} size="big" variant="outlined" className={styles.item}/>
+            <DisplayAction actionKey="pageComposer" context={{path: editPath}} render={getButtonRenderer({enabled: enabled})} size="big" variant="ghost" color="accent" className={styles.item}/>
```
Comment (tdraier) [inline src/javascript/JContent/ContentRoute/MainActionBar/MainActionBar.jsx]: Can you move out pageComposer action page-composer module ?

File: src/javascript/JContent/ContentRoute/MainActionBar/MainActionBar.jsx [hunk 3]
```diff
@@ -5,18 +5,18 @@ import {DisplayAction} from '@jahia/ui-extender';
...
+            <DisplayAction actionKey="edit" context={{path: editPath, enabled: enabled}} render={getButtonRenderer({enabled: enabled})} size="big" variant="outlined" className={styles.item}/>
```
Comment (tdraier) [inline src/javascript/JContent/ContentRoute/MainActionBar/MainActionBar.jsx]: Does it work if content-editor / page-composer  are stopped ?

Comment (tdraier) [inline src/javascript/JContent/ContentRoute/MainActionBar/MainActionBar.jsx]: Enabled status should be handled by the action

File: src/javascript/JContent/ContentRoute/MainActionBar/MainActionBar.jsx (hunk 1 above)
Comment (Yousria) [reply src/javascript/JContent/ContentRoute/MainActionBar/MainActionBar.jsx]: I need to have the selection to disable the buttons https://jira.jahia.org/browse/BACKLOG-12329 as it's described in the story

File: src/javascript/JContent/ContentRoute/MainActionBar/MainActionBar.jsx (hunk 2 above)
Comment (Yousria) [reply src/javascript/JContent/ContentRoute/MainActionBar/MainActionBar.jsx]: yes but then I'll not have access to the selection

File: src/javascript/JContent/ContentRoute/MainActionBar/MainActionBar.jsx (hunk 3 above)
Comment (Yousria) [reply src/javascript/JContent/ContentRoute/MainActionBar/MainActionBar.jsx]: are they supposed to be stopped ? I'll check that

Comment (Yousria) [reply src/javascript/JContent/ContentRoute/MainActionBar/MainActionBar.jsx]: I know but I don't have access to the selection from content-editor side or page-composer side

**Pull request closed without merging**